pandas/_libs/groupby.pyx

```diff
@@ -101,7 +101,11 @@ cdef float64_t median_linear_mask(float64_t* a, int n, uint8_t* mask) noexcept n
     return result
 
 
-cdef float64_t median_linear(float64_t* a, int n) noexcept nogil:
+cdef float64_t median_linear(
+    float64_t* a,
+    int n,
```

Comment:
FYI these should be `Py_ssize_t` not `int`. `int` is limited to `2 ** 31 - 1` on most platforms, so this would cause undefined behavior if ever called with an `n` that exceeded that (probably unlikely within groupby...but still possible)

Happy to have that in a follow up PR